src/lbfgs.jl

```diff
@@ -16,6 +16,9 @@ mutable struct LBFGSData{T, I <: Integer}
   b::Vector{Vector{T}}
   insert::I
   Ax::Vector{T}
+  shifted_p::AbstractArray{T} # Temporary matrix used in the computation solve_shifted_system!
```

Comment:
```suggestion
  shifted_p::Matrix{T} # Temporary matrix used in the computation solve_shifted_system!
```
You shouldn’t have fields with abstract types inside a struct.